process/caller.py

```diff
@@ -41,7 +43,36 @@ def check_agreement(
         :return: whether values agree or not
         :rtype: bool
         """
-        return np.allclose(previous, current, rtol=1.0e-6)
+        # Check for same shape: mfile length can change between iterations
+        if isinstance(previous, float) or previous.shape == current.shape:
+            return np.allclose(previous, current, rtol=1.0e-6, equal_nan=True)
+        else:
+            return False
+
+    @staticmethod
+    def check_mfile_agreement(
```

Comment:
Can this method replace the `check_agreement` method too? We seem to have two very similar static methods used at different points in the code.